spot-contracts/contracts/_utils/BondTranchesHelpers.sol

```diff
@@ -51,37 +53,23 @@ library BondTranchesHelpers {
         pure
         returns (uint256[] memory)
     {
-        uint256[] memory redeemableAmts = new uint256[](bt.tranches.length);
+        // NOTE: This implementation assumes the bond has only two tranches.
+        if (bt.tranches.length != 2) {
+            revert UnacceptableTrancheLength();
+        }
 
-        // We Calculate how many underlying assets could be redeemed from each available tranche balance,
-        // assuming other tranches are not an issue, and record the smallest amount.
-        //
-        // Usually one tranche balance is the limiting factor, we first loop through to identify
-        // it by figuring out the one which has the least `trancheBalance/trancheRatio`.
-        //
-        uint256 minBalanceToTrancheRatio = type(uint256).max;
-        uint8 i;
-        for (i = 0; i < bt.tranches.length; i++) {
-            // NOTE: We round the avaiable balance down to the nearest multiple of the
-            //       tranche ratio. This ensures that `minBalanceToTrancheRatio`
-            //       can be represented without loss as a fixedPt number.
-            uint256 bal = trancheBalsAvailable[i] - (trancheBalsAvailable[i] % bt.trancheRatios[i]);
-            uint256 d = bal.mulDiv(TRANCHE_RATIO_GRANULARITY, bt.trancheRatios[i]);
-            if (d < minBalanceToTrancheRatio) {
-                minBalanceToTrancheRatio = d;
-            }
+        uint256[] memory trancheAmtsReq = new uint256[](2);
 
-            // if one of the balances is zero, we return
-            if (minBalanceToTrancheRatio <= 0) {
-                return (redeemableAmts);
-            }
-        }
+        // We compute the amount of juniors required using all the seniors
```

Comment:
Does it make sense to first do a simple test, then an if-then?
```
if (srBal < jrBal * (srTR / jrTR) )) {
// seniors are the limiter, or correct ratio
...
} else {
// juniors are the limiter
...
}
```

Might depend on which limitation we expect to be most common. Your version should perform better when Jrs are less likely to be the limiter. The conditional check is cheap, but on failure it performs the calculations twice.

Reply:
This function is called from the vault and juniors are more likely to be the limiter

Reply:
This function is called from the vault, where we can assume that most times the seniors are the limiter (ie vault will have more juniors).